gcs: fix regression with input wizard "freezing"

mlyle opened the pull request:
Fixes #407

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/d-ronin/dronin/416)

<!-- Reviewable:end -->

mlyle committed:
gcs: input wizard: Progressively "center" switches as we calibrate

In case they were not centered, prevents the input wizard from getting
stuck in failsafe.  (This needs further work).

mlyle committed:
gcs: input wizard: display flight modes >=2 as 2

(rather than undefined)

mlyle committed:
gcs: input wizard: prevent range failsafe during setup

mlyle commented:
I still need to test this.  Just staging this here.

mlyle commented:
Jenkins build artifacts please.

tracernz commented on ground/gcs/src/plugins/config/configinputwidget.cpp:
```diff
@@ -595,12 +600,11 @@ void ConfigInputWidget::wizardSetUpStep(enum wizardSteps step)
             if(manualSettingsData.ChannelMin[i] <= manualSettingsData.ChannelMax[i]) {
-                manualSettingsData.ChannelMin[i]=manualSettingsData.ChannelNeutral[i];
-                manualSettingsData.ChannelMax[i]=manualSettingsData.ChannelNeutral[i];
+                manualSettingsData.ChannelMin[i]=manualSettingsData.ChannelNeutral[i] - INITIAL_OFFSET;
```
Spacing around equals in these lines.

mlyle commented:
jenkins build artifacts please

mlyle committed:
gcs: inputwidget: adjust code style/spacing

pug398 commented:
Tested and flown with inverted sbus. LGTM.

mlyle commented:
@pug398 Cool, thanks.  :)   I'm going to try and play with it a little and then hopefully this can speedily merge.

jihlein commented:
Successfully tested on SciSky Naze32, no configuration wizard issues.

Successfully tested Sparky with D4R-II PPM/FrSky Senor Hub, no configuration wizard issues,

GCS running on Win7 Laptop.

mlyle commented:
Tested all the edge cases, etc.  It does reasonably.  Doesn't manifest the bug.

dustin commented:
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[ground/gcs/src/plugins/config/configinputwidget.cpp, line 1022 \[r2\]](https://reviewable.io:443/reviews/d-ronin/dronin/416#-K7eH8bU9BnfnRevxmiQ:-K7eH8bU9BnfnRevxmiR:-635496646) ([raw file](https://github.com/d-ronin/dronin/blob/0e4031d2ed96857da6f699349647fd6b3678a34f/ground/gcs/src/plugins/config/configinputwidget.cpp#L1022)):*
I don't get the default here.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/d-ronin/dronin/416)*
<!-- Sent from Reviewable.io -->

mlyle commented:
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[ground/gcs/src/plugins/config/configinputwidget.cpp, line 1022 \[r2\]](https://reviewable.io:443/reviews/d-ronin/dronin/416#-K7eH8bU9BnfnRevxmiQ:-K7eHmcPAGmNlfONntpQ:1227148622) ([raw file](https://github.com/d-ronin/dronin/blob/0e4031d2ed96857da6f699349647fd6b3678a34f/ground/gcs/src/plugins/config/configinputwidget.cpp#L1022)):*
There was special purpose code to "center" the throttle channel properly.  I added it for switches.  Rather than an if/else tree, I handled those 3 cases (throttle, arming, flight mode).

Compilers often warn if you switch on an enum and don't handle all potential values, so I added a default.  break breaks out of the switch statement, not the for loop.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/d-ronin/dronin/416)*
<!-- Sent from Reviewable.io -->

dustin commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/d-ronin/dronin/416#-:-K7eILPdBGCbctasEQNb:1417088721)*
<!-- Sent from Reviewable.io -->
